sim_pkg/__main__.py

```diff
@@ -6,85 +8,37 @@
 import socketserver
 import csv
 import numpy as np
-from init_pose import init
 import argparse
+from sim_pkg.config import Configuration
 
-# sim_process_ = 0 
 robot_processes_ = []
-# vis_processes_ = 0
 parser = argparse.ArgumentParser(description='Run the simulator')
 
-parser.add_argument('-fn', '--filename', type=str, help="Name of the file without the .py extension")
+parser.add_argument('-fn', '--filename', type=str,
+                    help="Name of the file without the .py extension")
 
 args = parser.parse_args()
 
-def write_to_csv(free_port):
-    """
-    Write avaliable port number to csv file
-    """
-    f = open('port.csv', 'w')
-    num = [free_port]
-    with f:
-        # create the csv writer
-        writer = csv.writer(f)
-
-        # write a row to the csv file
-        writer.writerow(num)
-
-    # close the file
-    f.close()
-
-def init_pos(swarmsize):
-    """
-    Initialize the position
-    """
-    x = [0]*swarmsize
-    y = [0]*swarmsize
-    theta = [0]*swarmsize
-    a_ids = [0]*swarmsize
-
-    x, y, theta, a_ids = init(swarmsize,x,y,theta,a_ids)
-    id = [i for i in range(swarmsize)]
-
-    data = np.array([id, x, y, theta, a_ids]).T
-    np.savetxt("init.csv", data, delimiter=",", fmt="%.2f",
-           header="ID, x, y, theta, a_ids")
-
-
 def run():
     """
     Opens all the programs involved
     """
 
-    # Finds the first available socket on the system
-    with socketserver.TCPServer(("localhost", 0), None) as s:
```

Comment:
I am not comfortable with this. As a user, I do not want your process to be snooping around my computer searching for open ports. This is also doesn't *feel* super reliable due to how janky it is. As a user I'd much more prefer to just use my own port in `config.json`

Reply:
Ah now I see why you do this. Well, the problem you're having is that `OSError: [Errno 98] Address is already in use` is thrown, so you thought you'd hack around that. Cute

But, this is not the solution. You keep opening my ports. I hate that. Close the port you opened rather than keep opening new ports!!! I killed my `run.py` any my `simulator` is still running. Please fix that.

The way you've done it like this won't work. Traditionally it's done [with pid files](https://stackoverflow.com/questions/8296170/what-is-a-pid-file-and-what-does-it-contain)